cmsl1t/plotting/resolution_vs_X.py

```diff
@@ -20,44 +23,87 @@ def __init__(self, resolution_type, online_name, offline_name, versus_name):
         self.versus_name = versus_name
         self.resolution_method = get_resolution_function(resolution_type)
 
-    def create_histograms(self,
-                          online_title, offline_title, versus_title,
-                          pileup_bins, res_n_bins, res_low, res_high,
-                          vs_n_bins, vs_low, vs_high,
-                          ):
+    def create_histograms(
+        self, online_title, offline_title, versus_title, pileup_bins,
+        res_n_bins, res_low, res_high, vs_n_bins, vs_low, vs_high,
+    ):
         """ This is not in an init function so that we can by-pass this in the
         case where we reload things from disk """
         self.online_title = online_title
         self.offline_title = offline_title
         self.versus_title = versus_title
+        self.ymin = res_low
+        self.ymax = res_high
         self.pileup_bins = bn.Sorted(pileup_bins, "pileup",
                                      use_everything_bin=True)
 
-        name = ["resolution_vs", self.versus_name, self.online_name, self.offline_name, "pu_{pileup}"]
-        name = "__".join(name)
-        title = " ".join(["Resolution (", self.online_name, "vs.", self.offline_name,
-                          ") against", self.versus_name, "in PU bin: {pileup}"])
+        nameTokens = [
+            "resolution_vs", self.versus_name,
+            self.online_name, self.offline_name, "pu_{pileup}",
+        ]
+        name = "__".join(nameTokens)
+        title = 'Resolution ({online_name} vs. {offline_name}) '
+        title += 'against {versus_name}'
+        title = title.format(
+            online_name=self.online_name,
+            offline_name=self.offline_name,
+            versus_name=self.versus_name,
+        )
+        title += " in PU bin: {pileup}"
         title = ";".join([title, self.offline_title, self.online_title])
-        self.plots = HistogramCollection([self.pileup_bins],
-                                         "Hist2D", vs_n_bins, vs_low, vs_high,
-                                         res_n_bins, res_low, res_high,
-                                         name=name, title=title)
+        self.plots = HistogramCollection(
+            [self.pileup_bins],
+            "Hist2D", vs_n_bins, vs_low, vs_high,
+            res_n_bins, res_low, res_high,
+            name=name, title=title)
         self.filename_format = name
 
     def fill(self, pileup, versus, offline, online):
-        difference = self.resolution_method(online, offline)
+        difference = self.resolution_method(online, offline) + 1.
```

Comment:
Why do we add 1?  This feels like a quick-fix kludge that's been left behind.

Reply:
I think Alex mentioned something about this, and the point was that for energies (when `self.resolution_type == "energy"`) resolution_method will calculate `(online - offline)/offline)`. In that case, adding 1 becomes equivalent to `online / offline`, which is what we wanted here if I remember rightly.  But the screwy part is if this is _not_ used on energy quantities then adding 1 doesn't make sense, since it could mean doing something like `online - offline +1`.  Long story short, this should probably just be `online / offline` in this part of the code for now.

Reply:
removed